### PR TITLE
feat(storage-browser): add default elements

### PR DESCRIPTION
**Pull request by AllanZhengYP**
#### Description of changes

Move default elements for storage browser from examples to the component.

#### Description of how you validated changes
Screenshot 2024-08-16 at 4 09 59 PM

#### Checklist

- [x] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [x] PR description included
- [x] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/context/elements/defaults.tsx [hunk 1]
```diff
@@ -15,9 +15,13 @@ import {
   View as _View,
   ViewProps,
 } from '@aws-amplify/ui-react';
+import { ButtonElementProps, LabelElement, TextElement } from './definitions';
 
-const Button = React.forwardRef<HTMLButtonElement>(function Button(props, ref) {
-  const { disabled, variant } = props as any;
+const Button = React.forwardRef<HTMLButtonElement>(function Button(
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/context/elements/defaults.tsx:20]: Can provide `ButtonElementProps` as the second positional generic of `React.forwardRef` and remove the typing on lines 21 and 22 (same feedback for `Label`)

**Review by calebpollman — COMMENTED** [review]
Thanks for doing this! Minor typing feedback, and need to update the behavior added to `createProvider` to prevent bundle size impact

**Review by hbuchel — COMMENTED** [review]
Just a couple comments about missing exports.

File: packages/react-storage/src/components/StorageBrowser/index.ts [hunk 2]
```diff
@@ -11,3 +11,4 @@ export {
   CreateManagedAuthAdapterInput,
   StorageBrowserAuthAdapter,
 } from './adapters';
+export { elementsDefault } from './context/elements';
```
[the remark this answers is not in the document]
Comment (AllanZhengYP) [reply packages/react-storage/src/components/StorageBrowser/index.ts:14]: Exporting the `elementsDefault` will confuse the user that's for all Storage components, but it's only for StorageBrowser. 

I checked with @calebpollman that the `StorageBrowser` will eventually be exported via a subpath. We can take a followup task to export it in the StorageBrowser subpath.

[the remark this answers is not in the document]
Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/index.ts:14]: For this PR let's just export from the top level _index.ts_ so the migration doesn't impact inflight UI dev. All the exports will need to be updated with the subpath export addition regardless

File: packages/react-storage/src/components/StorageBrowser/context/elements/__tests__/defaults.spec.tsx [hunk 3]
```diff
@@ -1,9 +1,34 @@
...
+      region: 'region',
+      registerAuthListener: jest.fn(),
+    };
+    const { StorageBrowser } = createStorageBrowser({
+      elements: elementsDefault,
+      config,
+    });
+
+    render(<StorageBrowser />);
+    await waitFor(() => {
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/context/elements/__tests__/defaults.spec.tsx:24]: Are we sure the tests inside `waitFor` are actually passing or just that they are not throwing?

Comment (AllanZhengYP) [reply packages/react-storage/src/components/StorageBrowser/context/elements/__tests__/defaults.spec.tsx:24]: They are passing. Should I use `findBy*` instead?

Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/context/elements/__tests__/defaults.spec.tsx:24]: If the assertions fail inside `waitFor` the test will still pass. Ideally should be able to wrap the render call inside `waitFor` and run the assertions outside

**Review by calebpollman — APPROVED** [review]

**Comment by calebpollman** [comment]
🚢

**Review by hbuchel — APPROVED** [review]

**Pull request merged**